### PR TITLE
Rename akka package to org.apache.pekko

### PR DESCRIPTION
**Pull request by mdedetrich**
Current in progress PR

### CODE REVIEW

File: .scalafix.conf [hunk 1]
```diff
@@ -46,5 +46,5 @@ SortImports.blocks = [
   "scala.",
   "*",
   "com.sun."
-  "akka."
+  "org.apache.conf."
```
Comment (pjfanning) [inline .scalafix.conf:49]: should this be "org.apache.pekko."? Could you keep "akka." too - because pekko-http and other repos like that may keep their akka dependencies for a small period of time - until we have regular nightly jars of the core pekko modules.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala [hunk 2]
```diff
@@ -11,13 +11,14 @@ import scala.compat.java8.OptionConverters._
 import org.slf4j.Marker
 import org.slf4j.event.Level
 
-import akka.annotation.InternalApi
-import akka.util.OptionVal
+import org.apache.pekko
+import pekko.annotation.InternalApi
```
Comment (pjfanning) [inline akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: I'd prefer to not import org.apache.pekko and to fully qualify the imports like pekko.annotation.InternalApi.

IDEs like IntelliJ issue warnings about pekko.annotation.InternalApi style imports even though they are valid.

File: .scalafix.conf (hunk 1 above)
Comment (mdedetrich) [reply .scalafix.conf:49]: > should this be "org.apache.pekko."? 

Thanks for pickup, done

> Could you keep "akka." too - because pekko-http and other repos like that may keep their akka dependencies for a small period of time - until we have regular nightly jars of the core pekko modules.

This has nothing to do with dependencies, its used by scalafix to sort imports and none of the imports in our code will ever have `akka` in it (thats actually what this PR changes).

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala (hunk 2 above)
Comment (mdedetrich) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: > I'd prefer to not import org.apache.pekko and to fully qualify the imports like pekko.annotation.InternalApi.

I did it this way because it significantly reduces the diff and it also improves the quality of the improved documentation. In some places if you use the FQN it creates so much extra boilerplace that it also changes the scalafmt

> IDEs like IntelliJ issue warnings about pekko.annotation.InternalApi style imports even though they are valid.

At least on my machine Intellij has never produced such a warning.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestInbox.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import pekko.annotation.{ ApiMayChange, DoNotInherit }
+
+@ApiMayChange
+object TestInbox {
+  def apply[T](name: String = "inbox"): TestInbox[T] = {
+    val uid = ThreadLocalRandom.current().nextInt()
+    new TestInboxImpl((address / name).withUid(uid))
+  }
+
+  private[pekko] val address = RootActorPath(Address("akka.actor.typed.inbox", "anonymous"))
```
Comment (jxnu-liguobin) [inline akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestInbox.scala:24]: Does this need to be changed?

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+  /**
+   * ActorRef for this TestProbe
+   */
+  def ref: ActorRef[M]
+
+  /**
+   * Obtain time remaining for execution of the innermost enclosing `within`
+   * block or missing that it returns the properly dilated default for this
+   * case from settings (key "akka.actor.testkit.typed.single-expect-default").
```
Comment (jxnu-liguobin) [inline akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala:74]: Continue to use akka configuration?

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala (hunk 2 above)
Comment (He-Pin) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: I think we just keep it as FQCN, as spark and flink project is doing the same.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestInbox.scala (hunk 3 above)
Comment (mdedetrich) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestInbox.scala:24]: From what I have seen so far no, there are still some tests that need to be fixed but the actor address doesn't seem to effect anything.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala (hunk 4 above)
Comment (mdedetrich) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala:74]: Yes, this will be done in a future PR.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala (hunk 2 above)
Comment (mdedetrich) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: Flink is irrelevant because its largely a fully Java codebase now where you have to use FQCN and Spark doesn't use paradox to generate documentation from existing Scala code. As I said before, the bigger issue is that significant portion of Akka codebase uses the `akka.<SOMETHING>` package shorthand in return types. Changing that to `org.apache.pekko.<SOMETHING>` does significantly increase the codesize (in other words its not just imports). It would also make cherry picks from Ligthbends Akka less trivial.

I am not completely against using a FQCN but I would strongly recommend changing it in a future PR, not this one and possibly only for 1.1.x. Changing to use FQCN right now would take a really long amount of time (these changes already took a long time)

**Comment by He-Pin** [comment]
What about akka remote wire protocol and akka persistent?I think there is no way to do rolling migrating from akka cluster right?

**Comment by mdedetrich** [comment]
> What about akka remote wire protocol and akka persistent?I think there is no way to do rolling migrating from akka cluster right?

This will also be done in a future PR. This PR is **only** about changing the package name and any other relevant code changes that are necessary so the build doesn't fail. Also do note that the reason I created the draft PR is so I can run tests against the CI, its not ready and I need to update the original PR once its ready.

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala (hunk 4 above)
Comment (jxnu-liguobin) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/TestProbe.scala:74]: Yes, opening it on Mac M1 pro will cause Chrome to crash. 🥲

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala (hunk 2 above)
Comment (jrudolph) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: I'd say ultimately, we'd want fully-qualified imports because that's what everyone uses and also what IDEs usually create.

Regarding, the usages in code directly, I agree with your approach, it doesn't make sense to use FQCN in that case.

In any case, if you made significant changes and it works like this, cleaning up the imports can be done later, indeed. It will unfortunately lead to lots of conflicts so we should do it rather soon, though.

**Comment by jlprat** [comment]
@mdedetrich Do you want to change Scaladoc fully qualified references in this PR? 
There are 18 occurrences. For example: https://github.com/mdedetrich/incubator-pekko/blob/rename-akka-package-to-org-apache-pekko/akka-actor/src/main/java/org/apache/pekko/japi/pf/DeciderBuilder.java#L10

File: akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala (hunk 2 above)
Comment (mdedetrich) [reply akka-actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/CapturedLogEvent.scala:15]: If there is already consensus on this I can have a look at it, but I will do so after everything else is done.

**Comment by mdedetrich** [comment]
@jlprat I fixed the fully qualified references

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -263,7 +263,7 @@ in the file that describes briefly why the incompatibility can be ignored.
 
 Situations when it may be acceptable to ignore a MiMa issued warning include:
 
-- if it is touching any class marked as `private[akka]`, `/** INTERNAL API*/` or similar markers
+- if it is touching any class marked as `private[pekko]`, `/** INTERNAL API*/` or similar markers
```
Comment (He-Pin) [inline CONTRIBUTING.md:266]: does this still applied? private[org.apache.pekko]?

Comment (mdedetrich) [reply CONTRIBUTING.md:266]: Can you rephrase, I don't understand?

Comment (mdedetrich) [reply CONTRIBUTING.md:266]: Ah I understand, so technically speaking the source file does have `private[pekko]`, not `private[org.apache.pekko]` for the reasons stated earlier, so using `private[pekko]` is correct.

Comment (He-Pin) [reply CONTRIBUTING.md:266]: Ouch, you are right.

**Comment by mdedetrich** [comment]
Wahoo, so I have finally managed to get the CI to build however there are still some final steps left over

1. Need to update all documentation/comment references to old `akka.*SomeClass` to `org.apache.pekko.*SomeClass`
2. Paradox needs to be fixed, although it does statically check hard links to classes a **LOT** of the paradox documentation uses manual anchor/hyperlink references which are not statically checked, these need to be updated (and I also need to check the generated documentation doesn't have broken links).

**Comment by mdedetrich** [comment]
PR is now ready to review

**Comment by He-Pin** [comment]
I will check this on this weekend, great work.

**Review by pjfanning — APPROVED** [review]
Reviewed using Github Desktop tool.
Changes look good - thanks @mdedetrich 

With a PR this size, it is impossible to check every line in the diff but I think any issues can be fixed later. Key thing is that the CI still works.

**Comment by He-Pin** [comment]
I checked it locally with IDEA, so LGTM.
TODO:
1. config reference is till `akka` need to be updated to `pekko`.
2. project fold is still `akka` prefixed 

I think those two can comes in later PRs

**Review by He-Pin — APPROVED** [review]
I checked it locally with IDEA, so LGTM.
TODO:

config reference is till akka need to be updated to pekko.
project fold is still akka prefixed
I think those two can comes in later PRs

**Comment by mdedetrich** [comment]
Thanks for the reviews, I will do one last sanity check before merging it into main. 

> TODO:

>config reference is till akka need to be updated to pekko.
>project fold is still akka prefixed
>I think those two can comes in later PRs

Yes this is correct, we will do it in future PR's.

**Review by gmethvin — APPROVED** [review]
LGTM.

I checked out locally and looked over most of the changes, which seem correct, and the CI still works so I'm convinced this is probably fine.

**Comment by mdedetrich** [comment]
So I am going to merge the PR now, I noticed some more discrepancies in the paradox documentation and after screening it a few times I think I got all of them out.

**Comment by pjfanning** [comment]
We won't be doing the package rename in other repos yet. We need to start publishing builds from this repo first. The Akka imports in other repos are a mix of classes from this repo and the downstream repo and it will be easier to change them altogether.
